app/labor/rate_limit_checker.rb

```diff
@@ -5,19 +5,20 @@ def initialize(user = nil)
   end
 
   class UploadRateLimitReached < StandardError; end
+  class ArticleLimitReached < StandardError; end
 
   def limit_by_situation(situation)
     result = case situation
              when "comment_creation"
                user.comments.where("created_at > ?", 30.seconds.ago).size > 9
              when "published_article_creation"
-               user.articles.published.where("created_at > ?", 30.seconds.ago).size > 9
+               user.articles.published.where("created_at > ?", 1.hour.ago).size > 3
              when "image_upload"
                Rails.cache.read("#{user.id}_image_upload").to_i > 9
              else
                false
              end
-    ping_admins if result == true
+    ping_admins_without_delay if result == true
```

Comment:
Is there a reason why you changed this to be inline instead of pinging in the background?

Reply:
Firstly thanks for responding etc, this is exactly what im looking for to better understand the codebase.

I changed this as `ping_admins` was causing the request to get added to `delayed_jobs`, then timeout. The front-end would hang with `Processing...` being displayed. By changing this it allowed the process to be continued and handled.

Reply:
Oooh I got it. You should probably run `rails jobs:work` in a separate terminal, so that jobs will be processed. The timeout you experienced is weird indeed.